runtimes/pythonrt/svc.go

```diff
@@ -0,0 +1,360 @@
+// gRPC server that accepts calls from the Python runner
+package pythonrt
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"time"
+
+	pb "go.autokitteh.dev/autokitteh/proto/gen/go/autokitteh/remote/v1"
+	"go.autokitteh.dev/autokitteh/sdk/sdkservices"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/reflection"
+	"google.golang.org/grpc/status"
+)
+
+type remoteSvc struct {
+	pb.UnimplementedWorkerServer
+
+	log       *zap.Logger
+	cbs       *sdkservices.RunCallbacks
+	runID     sdktypes.RunID
+	xid       sdktypes.ExecutorID
+	syscallFn sdktypes.Value
+	lis       net.Listener
+	srv       *grpc.Server
+	port      int
+	runner    pb.RunnerClient
+
+	// We need the right context to send to cbs
+	// It can be either the initial context (start of flow) or call context (current activity)
+	// We don't have nested activities
+	initialCtx context.Context
+	callCtx    context.Context
+
+	// One of these will signal end of execution
+	done chan *pb.DoneRequest
+}
+
+func newRemoteSvc(log *zap.Logger, cbs *sdkservices.RunCallbacks, runID sdktypes.RunID, xid sdktypes.ExecutorID, syscallFn sdktypes.Value) *remoteSvc {
+	svc := remoteSvc{
+		log:   log,
+		cbs:   cbs,
+		runID: runID,
+		xid:   xid,
+
+		// Buffered so gRPC handler won't get stuck
+		done: make(chan *pb.DoneRequest, 1),
+	}
+
+	if syscallFn.IsValid() {
+		svc.syscallFn = syscallFn
+	}
+
+	return &svc
+}
+
+func (s *remoteSvc) Health(context.Context, *pb.HealthRequest) (*pb.HealthResponse, error) {
+	return &pb.HealthResponse{}, nil
```

Comment:
I'm thinking about checking with temporal here if the workflow is still active

wdyt ?
let's say someone canceled the workflow, or there was a worker restart somehow, not sure